json-schema-converter/src/test/java/io/confluent/connect/json/JsonSchemaDataTest.java

```diff
@@ -1031,13 +1031,39 @@ public void testToConnectIntEnumInAllOfIsInvalid() {
     schemas.add(numberSchema);
     schemas.add(enumSchema);
     CombinedSchema schema = CombinedSchema.allOf(schemas).build();
-    Schema expectedSchema = new SchemaBuilder(Schema.Type.STRING).parameter(JSON_TYPE_ENUM, "")
-        .parameter(JSON_TYPE_ENUM + ".1", "1")
-        .parameter(JSON_TYPE_ENUM + ".2", "2")
-        .parameter(JSON_TYPE_ENUM + ".3", "3")
+    Schema expectedSchema = new SchemaBuilder(Schema.Type.FLOAT64).build();
+
+    checkNonObjectConversion(expectedSchema, 123.45, schema, DoubleNode.valueOf(123.45));
+  }
+
+  @Test
+  public void testToConnectUnionEnumInAllOfIsValid() {
+    NumberSchema firstSchema = NumberSchema.builder()
+        .requiresInteger(true)
+        .unprocessedProperties(Collections.singletonMap("connect.type", "int8"))
+        .build();
+    NumberSchema secondSchema = NumberSchema.builder()
+        .requiresInteger(true)
+        .unprocessedProperties(Collections.singletonMap("connect.type", "int16"))
+        .build();
+    CombinedSchema oneof = CombinedSchema.oneOf(ImmutableList.of(firstSchema, secondSchema))
         .build();
+    EnumSchema enumSchema = EnumSchema.builder()
+        .possibleValue(1)
+        .possibleValue(2)
+        .possibleValue(3)
+        .build();
+    List<org.everit.json.schema.Schema> schemas = new ArrayList<>();
+    schemas.add(oneof);
+    schemas.add(enumSchema);
+    CombinedSchema schema = CombinedSchema.allOf(schemas).build();
+    SchemaBuilder builder = SchemaBuilder.struct().name(JSON_TYPE_ONE_OF);
+    builder.field(JSON_TYPE_ONE_OF + ".field.0", Schema.OPTIONAL_INT8_SCHEMA);
+    builder.field(JSON_TYPE_ONE_OF + ".field.1", Schema.OPTIONAL_INT16_SCHEMA);
+    Schema expectedSchema = builder.build();
```

Comment:
Did you forget the 
```
EnumSchema enumSchema = EnumSchema.builder()
        .possibleValue(1)
        .possibleValue(2)
        .possibleValue(3)
        .build();
```
in the expectedSchema?

Reply:
Before we threw an exception for an enum that wasn't a string.  Now if the enum is for an int or a union, we just ignore the enumerated values and return the int/union.

Reply:
So if the schema is a union of (int, Enums of int) it just gets converted to union of ints? But how can we convert it back from Connect schema to the JSON schema?

Reply:
Good question.  Yes, it will be lossy when going from Connect to JSON Schema.  We don't always guarantee lossy-ness.  But at least it won't throw an error now.

Reply:
Ok. I think it would be good to mention this behavior in the DGS ticket then? So we will notice it when things related happen in the future.